lgn-worker/src/main.rs

```diff
@@ -140,57 +135,42 @@ async fn main() -> anyhow::Result<()> {
         );
     }));
 
-    let last_task_processed =
-        AtomicU64::new(SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs());
-
-    if let Err(err) = run(cli, mp2_requirement, last_task_processed).await {
+    if let Err(err) = run(cli).await {
         panic!("Worker exited due to an error: {err:?}")
     } else {
         Ok(())
     }
 }
 
-async fn run(
-    cli: Cli,
-    mp2_requirement: semver::VersionReq,
-    last_task_processed: AtomicU64,
-) -> Result<()> {
+async fn run(cli: Cli) -> anyhow::Result<()> {
+    let mp2_version = semver::Version::parse(verifiable_db::version())?;
+    let mp2_requirement = semver::VersionReq::parse(&format!("^{mp2_version}"))?;
     let version = env!("CARGO_PKG_VERSION");
-    info!("Starting worker. version: {}", version);
+
     let config = Config::load(cli.config);
     config.validate();
     debug!("Loaded configuration: {:?}", config);
 
     let span = span!(
         Level::INFO,
-        "Starting node",
+        "run",
         "worker" = config.avs.worker_id.to_string(),
         "issuer" = config.avs.issuer.to_string(),
         "version" = version,
         "class" = config.worker.instance_type.to_string(),
     );
     let _guard = span.enter();
```

Comment:
A span is not a log entry, gave a proper name for the span and added a single log line below instead of the many as before.